### ISSUE TITLE
Enable Ingress Authentication/Authorization using Rancher

### ISSUE THREAD

**Issue by leodotcloud** [rancher/rancher#19664]
Nginx Ingress currently supports the following annotations to provide authentication.

```
    nginx.ingress.kubernetes.io/auth-signin: https://<RancherServer>
    nginx.ingress.kubernetes.io/auth-url: https://<RancherServer>/v3/auth/verify
```

If we could use Rancher to provide authentication and authorization, it would be very useful.

Sample use case:
If I create an Ingress in a Project,
- I want only the project members to be able to access it.
- Or I want any valid user in the AD/LDAP to be able to access it.

**Comment by gabrielcalderon** [comment]
+1 this would be AWESOME.

**Comment by vincent99** [comment]
You can set this up yourself if you want with something like `oauth2_proxy`, but rancher is not an oauth provider (or a meta-provider between you and an oauth provider like GitHub).

**Comment by leodotcloud** [comment]
@vincent99 I agree, there a lot of third-party options (authelia.com). Integrating with Rancher would provide a much better user experience.

**Comment by Fez29** [comment]
Hello All.

Any plans to look into this? This would be a really useful functionality, I want to do it with Azure AD so I can use JWT bearer tokens for API calls through Nginx-ingress, I have managed to get browser access right using ouath2_proxy but struggling with getting JWT tokens right.

**Comment by sota0113** [comment]
+1
